### PR TITLE
Add flag syntax for setting boolean params through the CLI. Closes #76.

### LINKED ISSUE #76: Flag syntax for boolean params
At the moment, CLI params are passed with arguments of the form `key=value`, meaning that boolean parameters are set with e.g. `use_bias=True` or `use_bias=False`.

It would be nice if we could have the option of passing boolean arguments with `--use-bias` or `--no-use-bias` to set that parameter to `True` and `False` respectively.

### PR DESCRIPTION
**Pull request by AdamHillier**

### CODE REVIEW

File: zookeeper/core/cli.py [hunk 1]
```diff
@@ -7,6 +7,14 @@
 
 class ConfigParam(click.ParamType):
     def convert(self, str_value, param, ctx):
+        # Allow the syntax `--name_of_key` to set the boolean flag to `True`,
+        # and likewise `--no-name_of_key` to set the boolean flag to `False`.
+        if re.match("^--[\\w.]+$", str_value):
+            return str_value[2:], True
+        if re.match("^--no-[\\w.]+$", str_value):
+            return str_value[5:], False
```
Comment (lgeiger) [inline zookeeper/core/cli.py]: Isn't there something we could use from `click` to do this parsing?

Comment (AdamHillier) [reply zookeeper/core/cli.py]: It's possible we could get it to work. This PR uses `ignore_unknown_options` as mentioned [here](https://click.palletsprojects.com/en/5.x/advanced/#forwarding-unknown-options) but I'm not sure how that interacts with the variadic argument which by default picks up all unmatched strings passed in.

Comment (AdamHillier) [reply zookeeper/core/cli.py]: By which I mean, I know that this does work correctly but I'm not sure how a different solution where Click does the parsing would interact correctly.

Comment (lgeiger) [reply zookeeper/core/cli.py]: Makes sense :+1:

**Pull request merged**